source/tools/glkernel-cli/main.cpp

```diff
@@ -0,0 +1,213 @@
+#include <iostream>
+#include <string>
+#include <fstream>
+
+#include <cppassist/logging/logging.h>
+
+#include <glkernel/glkernel-version.h>
+
+#include <cppassist/cmdline/ArgumentParser.h>
+#include <cppassist/cmdline/CommandLineProgram.h>
+#include <cppassist/cmdline/CommandLineAction.h>
+#include <cppassist/cmdline/CommandLineOption.h>
+#include <cppassist/cmdline/CommandLineParameter.h>
+#include <cppassist/cmdline/CommandLineSwitch.h>
+
+#include "KernelGenerator.h"
+#include "AbstractKernelExporter.h"
+
+#include "JsonImporter.h"
+#include "JsonExporter.h"
+#include "PngExporter.h"
+
+
+std::string extractInputFormat(const std::string & inFileName) {
+    if (inFileName.find('.') == std::string::npos)
+    {
+        return "";
+    }
+    const auto inFileFormat = inFileName.substr(inFileName.find_last_of('.') + 1);
+    if (inFileFormat != "js" && inFileFormat != "json")
+    {
+        return "";
+    }
+    return inFileFormat;
+}
+
+std::string extractOutputFormat(const std::string & outFileName, const bool shouldConvert) {
+    if (outFileName.find('.') == std::string::npos)
```

Comment:
There is a doubled search for `'.'` in the string. Search it once (`outFileName.find_last_of('.')`), store the result and reuse it both times.